### PR TITLE
PHPC-703: Update build scripts for SSL defines and system crypto policy

### PR DESCRIPTION
**Pull request by jmikola**
https://jira.mongodb.org/browse/PHPC-703

### CODE REVIEW

File: config.m4 [hunk 1]
```diff
@@ -3,8 +3,8 @@ PHP_ARG_ENABLE(mongodb, whether to enable mongodb support,
 [  --enable-mongodb           Enable mongodb support])
 PHP_ARG_WITH(openssl-dir, OpenSSL dir for mongodb,
 [  --with-openssl-dir[=DIR]  openssl install prefix], yes, no)
-
-
+PHP_ARG_WITH(system-ciphers, whether to use system default cipher list instead of hardcoded value,
+[  --with-system-ciphers   OPENSSL: Use system default cipher list instead of hardcoded value], no, no)
```
Comment (jmikola) [inline config.m4]: This is the same argument name used in the OpenSSL extension's `config0.m4`. I assume these should be kept in sync for cases where the extension is compiled into PHP directly (just as our `openssl-dir` arg syncs up with the one used for the FTP extension).

File: config.w32 [hunk 2]
```diff
@@ -40,24 +40,26 @@ if (PHP_MONGODB != "no") {
   if (CHECK_LIB("ssleay32.lib", "mongodb", PHP_MONGODB) &&
       CHECK_LIB("libeay32.lib", "mongodb", PHP_MONGODB) &&
       CHECK_HEADER_ADD_INCLUDE("openssl/ssl.h", "CFLAGS_MONGOC")) {
-    AC_DEFINE("MONGOC_ENABLE_OPENSSL", 1);
-    AC_DEFINE("MONGOC_ENABLE_LIBCRYPTO", 1);
+    AC_DEFINE("MONGOC_ENABLE_SSL_OPENSSL", 1);
+    AC_DEFINE("MONGOC_ENABLE_CRYPTO_LIBCRYPTO", 1);
     AC_DEFINE("MONGOC_ENABLE_SSL", 1);
     AC_DEFINE("MONGOC_ENABLE_CRYPTO", 1);
+    AC_DEFINE("MONGOC_ENABLE_CRYPTO_SYSTEM_PROFILE", 1);
```
Comment (jmikola) [inline config.w32]: I noted that `CMakeLists.txt` (for Windows) in libmongoc always sets this to 1, so I've opted to do the same here.

**Comment by derickr** [comment]
LGTM - Travis disagrees though.

File: config.m4 (hunk 1 above)
Comment (jmikola) [inline config.m4]: cc @remicollet

Comment (remicollet) [inline config.m4]: Nice, make sense, thanks

**Pull request merged**